clickhouse-jdbc/src/main/java/com/clickhouse/jdbc/ResultSetImpl.java

```diff
@@ -1,981 +1,1236 @@
+        this.closed = false;
+    }
+
+    private void checkClosed() throws SQLException {
+        if (closed) {
+            throw new SQLException("ResultSet is closed.");
+        }
+    }
+
 
     @Override
     public boolean next() throws SQLException {
-        return false;
+        checkClosed();
+
+        Map<String, Object> currentRow = reader.next();
+        return currentRow != null;
     }
 
     @Override
     public void close() throws SQLException {
+        closed = true;
+        if (reader != null) {
+            reader = null;
+        }
 
+        if (response != null) {
+            try {
+                response.close();
+            } catch (Exception e) {
+                throw new SQLException(e);
+            }
+            response = null;
+        }
     }
 
     @Override
     public boolean wasNull() throws SQLException {
+        checkClosed();
         return false;
     }
 
     @Override
     public String getString(int columnIndex) throws SQLException {
-        return "";
+        checkClosed();
+        return reader.getString(columnIndex);
     }
 
     @Override
     public boolean getBoolean(int columnIndex) throws SQLException {
-        return false;
+        checkClosed();
+        return reader.getBoolean(columnIndex);
     }
 
     @Override
     public byte getByte(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getByte(columnIndex);
     }
 
     @Override
     public short getShort(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getShort(columnIndex);
     }
 
     @Override
     public int getInt(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getInteger(columnIndex);
     }
 
     @Override
     public long getLong(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getLong(columnIndex);
     }
 
     @Override
     public float getFloat(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getFloat(columnIndex);
     }
 
     @Override
     public double getDouble(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getDouble(columnIndex);
     }
 
     @Override
     public BigDecimal getBigDecimal(int columnIndex, int scale) throws SQLException {
-        return null;
+        checkClosed();
+        return reader.getBigDecimal(columnIndex);
     }
 
     @Override
     public byte[] getBytes(int columnIndex) throws SQLException {
-        return new byte[0];
+        checkClosed();
+        return reader.getByteArray(columnIndex);
     }
 
     @Override
     public Date getDate(int columnIndex) throws SQLException {
-        return null;
+        checkClosed();
+        LocalDate localDate = reader.getLocalDate(columnIndex);
```

Comment:
This I think we need to move to the reader. Would you please add `TODO:` for me here in the code?